rholang/examples/linking/packages/ConditionedToken.rho

```diff
@@ -0,0 +1,109 @@
+export ConditionedToken in {
+  import BalanceMap, IMapApi in {
+    new
+      IMapEmptyMap, IMapUpdated, IMapGetOrElse, //IMap Methods
+      mapApiCh, addCh, transferCh, balanceOfCh //BalanceMap methods
+    in {
+      IMapApi("EmptyMap", IMapEmptyMap) | IMapApi("GetOrElse", IMapGetOrElse) |
+      IMapApi("Updated", IMapUpdated) | BalanceMap(mapApiCh) | 
+      for(
+          mapApi <- mapApiCh; emptyMap <- IMapEmptyMap; getOrElse <- IMapGetOrElse;
+          updated <- IMapUpdated
+      ) {
+        mapApi("Transfer", transferCh) | mapApi("BalanceOf", balanceOfCh) |
+        mapApi("Add", addCh) |
+        for(add <- addCh; transfer <- transferCh; balanceOf <- balanceOfCh) {
+          contract ConditionedToken(initId, initCondition, totalSupply, return) = {
```

Comment:
Currently, this assumes the total supply is fixed. Variable supply can be simulated through moving tokens from the initial account to user-created ones (mint event) or transferring tokens to an account which can never be withdrawn from (burn event). However, we may want to consider generalizing this to allow variable supply in a more natural way.

Would be interested in know if there anyone has opinions on this.